casanovo/config.py

```
@@ -202,6 +197,19 @@ def validate_param(self, param: str, param_type: Callable) -> None:
             )
 
+    def validate_values(self) -> None:
+        """Verify that parameter values satisfy semantic constraints."""
+        raw_train_check_interval = self._user_config.get(
+            "train_check_interval", self._params["train_check_interval"]
+        )
+        if (
+            not isinstance(self._params["train_check_interval"], int)
+            or isinstance(raw_train_check_interval, bool)
+            or isinstance(raw_train_check_interval, float)
+            or self._params["train_check_interval"] <= 0
```

Ruff (0.15.7)

[warning] Multiple `isinstance` calls for `raw_train_check_interval`, merge into a single call

Merge `isinstance` calls for `raw_train_check_interval`

(SIM101)